Assets/Scripts/PlayerMovement.cs

```diff
@@ -194,6 +200,31 @@ void OnCollisionEnter2D(Collision2D collision)
             case "LightningCloud":
                 Physics2D.IgnoreCollision(collision.gameObject.GetComponent<Collider2D>(), GetComponent<Collider2D>());
                 break;
+            case "Demon":
+                Debug.Log("Hit by demon");
+                playerReceiver.TakeDamage(20);
+                break;
+            case "Fireball":
+                if (!fireProjectile.enabled)
+                {
+                    fireProjectile.enabled = true;
+                    fireProjectile.fireballsText.enabled = true;
+                    collision.gameObject.SetActive(false);
+                }
+                else
+                {
+                    fireProjectile.collectFireballs();
+                    collision.gameObject.SetActive(false);
+                }
+                if (instructions.Contains(collision.gameObject))
+                {
+                    DisplayText("Press Shift to shoot in player's direction", collision.gameObject);
+                }
+                break;
+            case "VolcanoBall":
+                Debug.Log("Hit by volcanoBall");
+                playerReceiver.TakeDamage(50);
```

Comment:
Can we differentiate Fireballs from Demon and Lava from Volcano. Currently they look same but damage is different